#! /bin/sh /usr/share/dpatch/dpatch-run
## 112-remove-nonexistent-prototypes-from-gen.h.dpatch by  <kmccarty@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Remove prototypes from gen.h of functions that don't exist in mathlib.
## DP: Because of the way cfortran works, their existence otherwise causes
## DP: linking failures in third-party code.

@DPATCH@
diff -urNad cernlib-2005.05.09.dfsg~/src/cfortran/gen.h cernlib-2005.05.09.dfsg/src/cfortran/gen.h
--- cernlib-2005.05.09.dfsg~/src/cfortran/gen.h	2006-05-16 10:35:41.400252625 -0400
+++ cernlib-2005.05.09.dfsg/src/cfortran/gen.h	2006-05-16 10:37:14.159452693 -0400
@@ -84,10 +84,10 @@
 
 PROTOCCALLSFFUN1(FLOAT,ALGAMA,algama,FLOAT)
 #define ALGAMA(A2)  CCALLSFFUN1(ALGAMA,algama,FLOAT,A2)
-
+/*
 PROTOCCALLSFFUN1(FLOAT,QLGAMA,qlgama,FLOAT)
 #define QLGAMA(A2)  CCALLSFFUN1(QLGAMA,qlgama,FLOAT,A2)
-
+*/
 PROTOCCALLSFFUN1(DOUBLE,DLGAMA,dlgama,DOUBLE)
 #define DLGAMA(A2)  CCALLSFFUN1(DLGAMA,dlgama,DOUBLE,A2)
 
@@ -141,10 +141,10 @@
 
 PROTOCCALLSFFUN5(DOUBLE,DCAUCH,dcauch,DOUBLE,DOUBLE,DOUBLE,DOUBLE,DOUBLE)
 #define DCAUCH(A2,A3,A4,A5,A6)  CCALLSFFUN5(DCAUCH,dcauch,DOUBLE,DOUBLE,DOUBLE,DOUBLE,DOUBLE,A2,A3,A4,A5,A6)
-
+/*
 PROTOCCALLSFFUN6(DOUBLE,CHEBQU,chebqu,DOUBLE,DOUBLE,DOUBLE,PDOUBLE,INT,DOUBLE)
 #define CHEBQU(A2,A3,A4,A5,A6,A7)  CCALLSFFUN6(CHEBQU,chebqu,DOUBLE,DOUBLE,DOUBLE,PDOUBLE,INT,DOUBLE,A2,A3,A4,A5,A6,A7)
-
+*/
 PROTOCCALLSFFUN2(FLOAT,CHISIN,chisin,FLOAT,INT)
 #define CHISIN(A2,A3)  CCALLSFFUN2(CHISIN,chisin,FLOAT,INT,A2,A3)
 
@@ -204,13 +204,13 @@
 
 PROTOCCALLSFFUN3(DOUBLE,DVDOT,dvdot,INT,DOUBLEV,DOUBLEV)
 #define DVDOT(A2,A3,A4)  CCALLSFFUN3(DVDOT,dvdot,INT,DOUBLEV,DOUBLEV,A2,A3,A4)
-
+/*
 PROTOCCALLSFFUN2(DOUBLE,E401BD,e401bd,DOUBLE,DOUBLE)
 #define E401BD(A2,A3)  CCALLSFFUN2(E401BD,e401bd,DOUBLE,DOUBLE,A2,A3)
 
 PROTOCCALLSFFUN2(FLOAT,ELIN1,elin1,FLOAT,FLOAT)
 #define ELIN1(A2,A3)  CCALLSFFUN2(ELIN1,elin1,FLOAT,FLOAT,A2,A3)
-
+*/
 PROTOCCALLSFFUN1(FLOAT,ERF,erf,FLOAT)
 #define ERF(A2)  CCALLSFFUN1(ERF,erf,FLOAT,A2)
 
@@ -237,13 +237,13 @@
 
 PROTOCCALLSFFUN2(FLOAT,FREARG,frearg,FLOAT,FLOAT)
 #define FREARG(A2,A3)  CCALLSFFUN2(FREARG,frearg,FLOAT,FLOAT,A2,A3)
-
+/*
 PROTOCCALLSFFUN3(FLOAT,FRED2,fred2,FLOAT,FLOAT,FLOAT)
 #define FRED2(A2,A3,A4)  CCALLSFFUN3(FRED2,fred2,FLOAT,FLOAT,FLOAT,A2,A3,A4)
 
 PROTOCCALLSFFUN1(FLOAT,FRED3,fred3,FLOAT)
 #define FRED3(A2)  CCALLSFFUN1(FRED3,fred3,FLOAT,A2)
-
+*/
 PROTOCCALLSFFUN1(FLOAT,FREQ,freq,FLOAT)
 #define FREQ(A2)  CCALLSFFUN1(FREQ,freq,FLOAT,A2)
 
@@ -270,10 +270,10 @@
 
 PROTOCCALLSFFUN1(FLOAT,GAMMA,gamma,FLOAT)
 #define GAMMA(A2)  CCALLSFFUN1(GAMMA,gamma,FLOAT,A2)
-
+/*
 PROTOCCALLSFFUN1(FLOAT,QGAMMA,qgamma,FLOAT)
 #define QGAMMA(A2)  CCALLSFFUN1(QGAMMA,qgamma,FLOAT,A2)
-
+*/
 PROTOCCALLSFFUN1(DOUBLE,DGAMMA,dgamma,DOUBLE)
 #define DGAMMA(A2)  CCALLSFFUN1(DGAMMA,dgamma,DOUBLE,A2)
 
@@ -297,16 +297,16 @@
 
 PROTOCCALLSFFUN4(FLOAT,GAUSS,gauss,ROUTINE,FLOAT,FLOAT,FLOAT)
 #define GAUSS(A2,A3,A4,A5)  CCALLSFFUN4(GAUSS,gauss,ROUTINE,FLOAT,FLOAT,FLOAT,A2,A3,A4,A5)
-
+/*
 PROTOCCALLSFFUN4(FLOAT,QGAUSS,qgauss,ROUTINE,FLOAT,FLOAT,FLOAT)
 #define QGAUSS(A2,A3,A4,A5)  CCALLSFFUN4(QGAUSS,qgauss,ROUTINE,FLOAT,FLOAT,FLOAT,A2,A3,A4,A5)
-
+*/
 PROTOCCALLSFFUN4(DOUBLE,DGAUSS,dgauss,ROUTINE,DOUBLE,DOUBLE,DOUBLE)
 #define DGAUSS(A2,A3,A4,A5)  CCALLSFFUN4(DGAUSS,dgauss,ROUTINE,DOUBLE,DOUBLE,DOUBLE,A2,A3,A4,A5)
-
+/*
 PROTOCCALLSFFUN6(DOUBLE,GPINDP,gpindp,DOUBLE,DOUBLE,DOUBLE,PDOUBLE,DOUBLE,INT)
 #define GPINDP(A2,A3,A4,A5,A6,A7)  CCALLSFFUN6(GPINDP,gpindp,DOUBLE,DOUBLE,DOUBLE,PDOUBLE,DOUBLE,INT,A2,A3,A4,A5,A6,A7)
-
+*/
 PROTOCCALLSFFUN2(INT,IGET,iget,INTV,INT)
 #define IGET(A2,A3)  CCALLSFFUN2(IGET,iget,INTV,INT,A2,A3)
 
@@ -321,10 +321,10 @@
 
 PROTOCCALLSFFUN3(INT,LKKSPL,lkkspl,DOUBLE,DOUBLEV,INT)
 #define LKKSPL(A2,A3,A4)  CCALLSFFUN3(LKKSPL,lkkspl,DOUBLE,DOUBLEV,INT,A2,A3,A4)
-
+/*
 PROTOCCALLSFFUN4(INT,LOCHAR,lochar,INTV,INT,INTV,INT)
 #define LOCHAR(A2,A3,A4,A5)  CCALLSFFUN4(LOCHAR,lochar,INTV,INT,INTV,INT,A2,A3,A4,A5)
-
+*/
 PROTOCCALLSFFUN2(INT,MAXFZE,maxfze,FLOATV,INT)
 #define MAXFZE(A2,A3)  CCALLSFFUN2(MAXFZE,maxfze,FLOATV,INT,A2,A3)
 
@@ -339,10 +339,10 @@
 
 PROTOCCALLSFFUN2(INT,NDIGIT,ndigit,DOUBLE,DOUBLE)
 #define NDIGIT(A2,A3)  CCALLSFFUN2(NDIGIT,ndigit,DOUBLE,DOUBLE,A2,A3)
-
+/*
 PROTOCCALLSFFUN1(INT,NEAR1,near1,FLOAT)
 #define NEAR1(A2)  CCALLSFFUN1(NEAR1,near1,FLOAT,A2)
-
+*/
 PROTOCCALLSFFUN1(INT,NIC311,nic311,FLOAT)
 #define NIC311(A2)  CCALLSFFUN1(NIC311,nic311,FLOAT,A2)
 
